docs/debugging-allocations.md

```diff
@@ -0,0 +1,187 @@
+```
+
+which is literally just the exact numbers of each of the 10 runs we considered. If you see a lot of fluctuation between the runs, then the allocation test doesn't allocate deterministically which would be something you want to fix. In the example above, we're totally stable to the last byte of allocated memory though.
+
+## Important notes
+
+The exact allocation counts are _not_ directly comparable between different operating systems and even different versions of the same operating system. Every new Swift version will also introduce differences (hopefully decreasing over time).
+
+This means that if you get hit by a CI failure, you will have to first reproduce the regression locally by running the allocation counter tests once on `master` and then again on your failing branch.
+
+## Debugging a regression in a particular test
+
+Because it takes quite a while to run the whole allocation counter test suite, there are some shortcuts you definitely want to know about. First, let's have a look at how we can run just single test.
+
+
+### Running a single test
+
+Running a single test is very straightforward, pass the test's `.swift` file to the runner script. For example:
+
+    ./run-nio-alloc-counter-tests.sh test_future_lots_of_callbacks.swift
+
+You will notice that it now runs faster but also that it still compiles SwiftNIO for every single time you run the program. So that's still rather slow. If you look closely at the output you will notice that it prints messages such as
+
+```
+Fetching /private/tmp/.nio_alloc_counter_tests_5jMMhk/swift-nio
+Fetching /private/tmp/.nio_alloc_counter_tests_5jMMhk/HookedFunctions
+Fetching /private/tmp/.nio_alloc_counter_tests_5jMMhk/AtomicCounter
+```
+
+If you change your working directory to common directory above (`/private/tmp/.nio_alloc_counter_tests_5jMMhk`), then you will find a regular Swift package there and you can just re-run the allocation counter tests using
+
+```
+cd /tmp/.nio_alloc_counter_tests_5jMMhk
+swift run -c release
+test_future_lots_of_callbacks.total_allocated_bytes: 4138056
```

Comment:
I don't think the test output is necessary here; the important info is `swift run -c release`

Reply:
good point, done